homeassistant/components/emoncms/__init__.py

```diff
@@ -21,6 +24,39 @@ async def async_setup_entry(hass: HomeAssistant, entry: EmonCMSConfigEntry) -> b
         entry.data[CONF_API_KEY],
         session=async_get_clientsession(hass),
     )
+    emoncms_unique_id = await emoncms_client.async_get_uuid()
+    if emoncms_unique_id:
+        if entry.unique_id != emoncms_unique_id:
+            ent_reg = er.async_get(hass)
+            entry_entities = ent_reg.entities.get_entries_for_config_entry_id(
+                entry.entry_id
+            )
+            for entity in entry_entities:
+                if entity.unique_id.split("-")[0] == entry.entry_id:
+                    feed_id = entity.unique_id.split("-")[-1]
+                    LOGGER.debug(f"moving feed {feed_id} to hardware uuid")
+                    ent_reg.async_update_entity(
+                        entity.entity_id, new_unique_id=f"{emoncms_unique_id}-{feed_id}"
+                    )
+            hass.config_entries.async_update_entry(
+                entry,
+                data=entry.data,
+                unique_id=emoncms_unique_id,
+            )
```

Comment:
You only need to update the unique_id, not the data
```suggestion
            hass.config_entries.async_update_entry(
                entry,
                unique_id=emoncms_unique_id,
            )
```